apps/ewallet/lib/ewallet/web/v1/serializers/transaction_request_serializer.ex

```diff
@@ -25,6 +25,7 @@ defmodule EWallet.Web.V1.TransactionRequestSerializer do
       user_id: transaction_request.user_id,
       account_id: transaction_request.account_id,
       require_confirmation: transaction_request.require_confirmation,
+      current_consumptions_count: length(transaction_request.consumptions),
```

Comment:
This preloads all consumption records then count them though. Can we do a count query?

```ex
current_consumptions_count: Repo.one(from c in TransactionConsumption, select: count(c.id), where: c.transaction_request_id == ^transaction_request.id)
```

Feel free to move the query to `TransactionRequest.count_consumptions(transaction_request)` or something like that :)

Reply:
@unnawut The `preload` will load everything in one request though (because we preload `minted_token` anyway) so I'm not sure it's worth it to do a separate count query. With that approach, we only have less stuff in memory. What do you think?

Reply:
Oh ok. Leave it as is then.